observer/probers/tls/tls.go

```diff
@@ -0,0 +1,193 @@
+package probers
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/crypto/ocsp"
+)
+
+type reason int
+
+type badOutcomeError struct {
+	reason reason
+}
+
+const (
+	none reason = iota
+	internalError
+	issuerVerifyFailed
+	ocspUnknown
+	ocspError
+)
+
+var reasonToString = map[reason]string{
+	none:               "nil",
+	internalError:      "internalError",
+	issuerVerifyFailed: "issuerVerifyFailed",
+	ocspUnknown:        "ocspUnknown",
+	ocspError:          "ocspError",
```

Comment:
If you like, these strings can be more human readable, like "got OCSP response 'unknown'". They're only going to show up in our logs, so being readable is a boon.